### ISSUE TITLE
Resettable switch option

### ISSUE THREAD

**Issue by Olivierbkk** [Plankske/hb-virtual-switch#4]
In the Homebridge dummy plugin, there is the option to make the switch resettable. Could you add this to your plugin?

![image](https://github.com/user-attachments/assets/b0688d4d-c7fb-4f3b-a933-ab7d6a762b9d)

**Comment by Plankske** [comment]
Let me check this out. Give me some time to look into this.

On Oct 5, 2024, at 7:57 PM, Olivierbkk ***@***.***> wrote:

﻿

In the Homebridge dummy plugin, there is the option to make the switch resettable. Could you add this to your plugin?

image.png (view on web)<https://github.com/user-attachments/assets/b0688d4d-c7fb-4f3b-a933-ab7d6a762b9d>

**Comment by Olivierbkk** [comment]
No rush, buddy... your plugin will certainly replace the HB dummy plugin but it's not really needed until HB2.0 is out

**Comment by Plankske** [comment]
To clarify - resettable means that if the timed switch is in its triggered state (say was turned ON for 2 minutes) is triggered again (say turned ON by some automation), do you want the timer to reset to 2 minutes again? If this is not what is understood by it, please clarify and I will see what I can do. If it is what I described above, it is active now in v.1.2.1  
Lmk.

**Comment by Olivierbkk** [comment]
Yes it’s correct. I tried it and it works, thanks!

I noticed you didn’t make this feature optional which is fine for my current use case but might not work for everyone else.

There is a bug to fix also, with persistent timer enabled, when I restart Homebridge, the switch is always on, even if it was off prior to restarting and doesn’t turn off even after the timer has elapsed. I have to manually turn it off and then it works as intended.
